lib/pasarguard-restore.sh

```
@@ -53,19 +73,335 @@ pg_backup_layout() {
+        [ -n "$dbname" ] || continue
+        if [ "$has_ts" = "1" ] && [ -n "$source_version" ] && [ "$source_version" != "$target_version" ]; then
+            needs_conversion=true
+            break
+        fi
+    done <"$manifest"
+    [ "$needs_conversion" = true ] || return 0
+
+    local pg_major=$((server_version_num / 10000))
+    local target_series="${target_version%.*}"
+    local compat_image="${requested_compat_image:-timescale/timescaledb-ha:pg${pg_major}-ts${target_series}-all}"
+    local compat_suffix="${$}-${RANDOM}-$(date +%s)"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)